crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -374,79 +375,35 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientState {
                 let keymap_state = state.keymap_state.as_ref().unwrap();
                 let key_utf8 = keymap_state.key_get_utf8(Keycode::from(key + MIN_KEYCODE));
                 let key_sym = keymap_state.key_get_one_sym(Keycode::from(key + MIN_KEYCODE));
-
-                let key = if matches!(
-                    key_sym,
-                    xkb::Keysym::BackSpace
-                        | xkb::Keysym::Left
-                        | xkb::Keysym::Right
-                        | xkb::Keysym::Down
-                        | xkb::Keysym::Up
-                        | xkb::Keysym::Super_L
-                        | xkb::Keysym::Super_R
-                ) {
-                    xkb::keysym_get_name(key_sym).to_lowercase()
-                } else {
-                    key_utf8.clone()
-                };
+                let key = xkb::keysym_get_name(key_sym).to_lowercase();
 
                 let focused_window = &state.keyboard_focused_window;
                 if let Some(focused_window) = focused_window {
                     match key_state {
                         wl_keyboard::KeyState::Pressed => {
-                            if key_sym == xkb::Keysym::Shift_L || key_sym == xkb::Keysym::Shift_R {
-                                state.modifiers.shift = true;
-                            } else if key_sym == xkb::Keysym::Control_L
-                                || key_sym == xkb::Keysym::Control_R
-                            {
-                                state.modifiers.control = true;
-                            } else if key_sym == xkb::Keysym::Alt_L || key_sym == xkb::Keysym::Alt_R
-                            {
-                                state.modifiers.alt = true;
-                            } else {
-                                focused_window.handle_input(KeyDown(KeyDownEvent {
-                                    keystroke: Keystroke {
-                                        modifiers: state.modifiers,
-                                        key,
-                                        ime_key: None,
-                                    },
-                                    is_held: false, // todo!(linux)
-                                }));
-                            }
+                            focused_window.handle_input(PlatformInput::KeyDown(KeyDownEvent {
+                                keystroke: Keystroke {
+                                    modifiers: state.modifiers,
+                                    key,
+                                    ime_key: Some(key_utf8),
```

Comment:
I've tested this PR a bit and it seems that unbinded `ctrl-...` shortcuts will still insert one garbage character. We should probably set `ime_key` to `None` if `key_utf8`'s length is made of a single non-printable character (`< 0x20`).

Reply:
Done